c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -900,15 +938,19 @@ public:
     auto up = factory.streamFactory.kjToCapnp(kj::mv(eofWrapper));
     pb.setUp(kj::cp(up));
 
-    context.setPipeline(pb.build());
-    context.initResults(capnp::MessageSize { 4, 1 }).setUp(kj::mv(up));
-
     auto response = kj::heap<HttpOverCapnpConnectResponseImpl>(
         factory, context.getParams().getContext());
 
-    return inner->connect(host, headers, *pipe.ends[0], *response, settings).attach(
+    auto result = inner->connect(host, headers, *pipe.ends[0], *response, settings).attach(
         kj::mv(host), kj::mv(headers), kj::mv(response), kj::mv(pipe))
         .exclusiveJoin(kj::mv(pumpTask));
+
+    auto tlsStarterImpl = kj::heap<TlsStarterImpl>(kj::mv(tlsStarter));
+    pb.setTlsStarter(kj::mv(tlsStarterImpl));
+
+    context.setPipeline(pb.build());
+    context.initResults(capnp::MessageSize { 4, 1 }).setUp(kj::mv(up));
```

Comment:
It's important that the final results match the pipeline set with `setPipeline`, so please initialize `tlsStarter` in the final results here.

Don't forget to change the cap count in `MessageSize` from 1 to 2.

Reply:
hm, I need to call `setTlsStarter` on both the pipeline and the final results right? Do I need a refcounted `TlsStarterImpl` or is there a better way? For now I resolved this by using a refcount.